### ISSUE TITLE
SerializeFieldSuppressor fails to suppress CS0649 when loaded from a VSIX or MPack

### ISSUE THREAD

**Issue by AnBucyk** [microsoft/Microsoft.Unity.Analyzers#15]
#### Bug description
SerializeFieldSuppressor fails to suppress CS0649 in supported scenarios.

- Version of analyzers assembly: Latest master
- Analyzer rule: USP0007
- Error:  `D:\Unity Projects\SerializeFieldReferenceTest\Assets\Scripts\SerializeReferenceTestScript.cs(8,20): warning CS0649: Field 'Foo.bar' is never assigned to, and will always have its default value null`

#### To Reproduce
Steps or code to reproduce the behavior:
1. Create empty GameObject, and attach a new script `Foo` to it.
2. Insert the following code into `Foo.cs`:

```csharp
using UnityEngine;

public class Foo : MonoBehaviour
{
    [SerializeField]
    private string bar;

    public void F()
    {
        Debug.Log(bar);
    }
}
```

3. Build
4. CS0649 appears in both the build output and in Visual Studio

#### Expected behavior
CS0649 is suppressed

#### Screenshots
![image](https://user-images.githubusercontent.com/12160725/73500803-8f727180-4378-11ea-924e-5682fa078dc8.png)

**Comment by sailro** [comment]
Do you repro when playing with this setting ? 

![image](https://user-images.githubusercontent.com/638167/73682888-19228780-4676-11ea-962b-7bcc7a5f4775.png)

(You need to close the solution / reload after changing this one)

**Comment by AnBucyk** [comment]
No, it doesn't repro unless you full build. @jbevain and I had issues reproing it on my machine until we did more than an incremental build.

**Comment by jbevain** [comment]
I spent quite some time looking at this over the weekend.

If you run the suppressors from a VSIX it's not going to get picked up.
If you run the suppressors from an analyzer DLL that is referenced in the csproj, which is the normal use case for Unity developers where VSTU references our suppressors, it's going to get picked up if you do a full build, but not in the lightweight build. I've reached out to the Roslyn team to figure out if we can solve for the VSIX scenario and for the lightweight build scenario.

**Comment by jbevain** [comment]
Update on this, we're looking at fixing the suppressors being picked up in the VSTU lightweight build in Visual Studio 16.6 and Visual Studio for Mac 8.6.

**Comment by aybe** [comment]
It also happens on things like this:

```
    internal abstract class Whatever : ScriptedImporter
    {
        public Transformation Transformation; // ScriptableObject here CS0649
    }
```

**Comment by jbevain** [comment]
@aybe good catch, we're not taking the `ScriptedImporter` type into consideration yet. I'll open a new bug for this one.

**Comment by jbevain** [comment]
I'm going to close this as this is a VS and VSTU issue and doesn't belong here.
